### DISCUSSION TITLE
v1.20

### DISCUSSION THREAD

**Discussion by duracelltomi** [duracelltomi/gtm4wp#322] [General]
THE BIG CLEANUP RELEASE!

Lots of deprecated features removed.
The code of WooCommerce integration was cleaned and restructured, if you have any custom code that relies on GTM4WP internal data structure,
please update the plugin on a test version of your website before updating your live site!

* Removed: deprecated Universal Analytics events for WooCommerce
* Removed: deprecated filter GTM4WP_WPFILTER_AFTER_DATALAYER/gtm4wp_after_datalayer
* Removed: deprecated filter GTM4WP_WPFILTER_ADDGLOBALVARS/gtm4wp_add_global_vars
* Removed: deprecated feature that puts the 'Do not track' option of the browser into the data layer. It is a deprecated browser feature therefore removed from this plugin.
* Removed: deprecated Google Optimize integration
* Removed: deprecated constants GTM4WP_OPTION_BLACKLIST_MACRO_* and GTM4WP_OPTION_BLACKLIST_<adsystemname>
* Removed: deprecated WordPress filter GTM4WP_WPFILTER_GETTHEGTMTAG (deprecated in v1.16)
* Changed: lots of internal function names and variable names changed, although is not recommended to depend on them, if you did, check your code
* Changed: renamed WP filter gtm4wp_datalayer_on_pageload to gtm4wp_woocommerce_datalayer_on_pageload to better reflect when it is called.
* Changed: orderData data layer variable in WooCommerce integration now includes keys to better support enhanced conversions setup:
  * orderData.customer.billing.first_name_hash
  * orderData.customer.billing.last_name_hash
  * orderData.customer.billing.phone_hash
* Deprecated: orderData.customer.billing.emailhash, please update to use orderData.customer.billing.email_hash (with an underscore)
* Added: stockstatus key into the product array of every ecommerce action. Returns the value of WP_Product->get_stock_status(). Thanks [hans2103](https://github.com/hans2103).
* Added: integration with WebToffee GDPR Cookie Consent plugin. GTM4WP can not fire a GTM event when user consent changes or when a previously stored consent has been loaded.
* Fixed: add_payment_info and add_shipping_info events were not fired during checkout submit when not fired before on the page.
* Fixed: GTM4WP will only look for the user's IP address in the REMOTE_ADDR server variable. You may enter a custom HTTP header instead in plugin settings.

<hr /><em>This discussion was created from the release <a href='https://github.com/duracelltomi/gtm4wp/releases/tag/1.20'>v1.20</a>.</em>

**Comment by rockyoptima** [comment]
GAds - Conversion Tag - Purchase is not fired, as a result, no purchase value on Google Ads.

**Reply by duracelltomi** [reply]
Do you have a public test version of your site where I can place a test order without the need to use a bank card?

**Reply by rockyoptima** [reply]
Thanks for promot reply but I do not have one.

**Reply by duracelltomi** [reply]
Unfortunately without being able to examine your order received page I will not be able to help :-(

**Reply by rockyoptima** [reply]
I am switch back to previous vertion for now. Should I also change the event names back to gtm4wp.XXXX?

**Reply by duracelltomi** [reply]
To track with GA4, you can keep using GA4 event names

**Reply by rockyoptima** [reply]
GA4 is working fine with 1.2 version, just Google Ads do not have any transaction value recorded.

**Reply by duracelltomi** [reply]
In that case the data layer is there.
Can you post screenshots about your GAds conversion tag and the trigger attached to it?

**Reply by rockyoptima** [reply]
![2024-02-07_17-12-56](https://github.com/duracelltomi/gtm4wp/assets/1959648/74159179-cca1-4425-865d-9047dbc3ec4f)

**Reply by duracelltomi** [reply]
Conversion label is empty?

**Reply by duracelltomi** [reply]
Also Conversion value has to be read from a different data layer variable

**Reply by duracelltomi** [reply]
https://gtm4wp.com/google-tag-manager-for-woocommerce/how-to-setup-google-ads-conversion-tracking-for-woocommerce

**Reply by rockyoptima** [reply]
it is not empty,I blured it, but some how it is shown as white.

**Reply by rockyoptima** [reply]
this will cause event not fired? or just conversion value can not be read correctly?

**Reply by duracelltomi** [reply]
I think the value is not tracked

**Reply by ipotion** [reply]
Hello,
Gtm4wp is properly working on my site but I need to implement consent (i added the built in consent checks)
 
About the  webtoofee integration. 
Is it enough to activate it to have Google Consent V2  working?
or there should additional code be placed in functions.php  like this one:
https://gist.github.com/webtoffee-git/9d7992d6afb329f4700a6984636695c2

Thank you!

**Reply by duracelltomi** [reply]
@ipotion no, it is not. The integration only fires a GTM event that you can use inside GTM to fire a consent tag with the update command.

**Reply by ipotion** [reply]
Simple for you :)) i feel in the dark here.

**Reply by duracelltomi** [reply]
What about this article? https://gtm4wp.com/setup-gtm4wp-features/webtoffee-gdpr-cookie-consent-plugin-gtm4wp-how-to-setup
